build-tools/src/main/kotlin/org/jetbrains/kotlin/CopySamples.kt

```diff
@@ -3,35 +3,53 @@ package org.jetbrains.kotlin
 import groovy.lang.Closure
 import org.gradle.api.Task
 import org.gradle.api.tasks.Copy
+import org.gradle.api.tasks.InputDirectory
+import java.io.File
 
 /**
  * A task that copies samples and replaces direct repository URLs with ones provided by the cache-redirector service.
+ * This task also adds kotlin compiler repository from the project's gradle.properties file.
  */
 open class CopySamples: Copy() {
-
-    var samplesDir = project.file("samples")
-
-    init {
-        configureReplacements()
-    }
-
-    fun samplesDir(path: Any) {
-        samplesDir = project.file(path)
-    }
+    @InputDirectory
+    var samplesDir: File = project.file("samples")
 
     private fun configureReplacements() {
         from(samplesDir) {
+            it.exclude("**/*.gradle.kts")
             it.exclude("**/*.gradle")
+            it.exclude("**/gradle.properties")
         }
         from(samplesDir) {
             it.include("**/*.gradle")
+            val replacements = replacementsWithWrapper { s -> "maven { url '$s' }" }
+            it.filter { line ->
+                val repo = line.trim()
+                replacements[repo]?.let { r -> line.replace(repo, r) } ?: line
+            }
+        }
+        from(samplesDir) {
+            it.include("**/*.gradle.kts")
+            val replacements = replacementsWithWrapper { s -> "maven(\"$s\")"}
             it.filter { line ->
-                replacements.forEach { (repo, replacement) ->
-                    if (line.contains(repo)) {
-                        return@filter line.replace(repo, replacement)
-                    }
+                val repo = line.trim()
+                replacements[repo]?.let { r -> line.replace(repo, r) } ?: line
+            }
+        }
+        from(samplesDir) {
```

Comment:
This logic works on CI only. But what if I want to run samples locally against a compiler built from sources? May be additionally bump Kotlin version in the samples to e.g. 1.4.20-RC?

This particular change seems good to me though. It is useful because it allows us to check samples against a fresh version of the Gradle plugin on CI.

Reply:
> May be additionally bump Kotlin version in the samples to e.g. 1.4.20-RC?
It bumps during the release, and it still won't work in case something have changed in kotlin-native repo, while latest RC/eap/M* doesn't have appropriate changes.

> But what if I want to run samples locally against a compiler built from sources?
If you build with a composite build you'll be able to run it locally, but it won't work for gradle plugin unless you have manually published it locally, then I guess this version could be changed to something like `1.4.30-SNAPSHOT-255`